tools/.golangci.yaml

```diff
@@ -17,6 +18,47 @@ linters:
     - usetesting
     - whitespace
   settings:
+    depguard:
+      rules:
+        api:
+          list-mode: lax
+          files:
+            - '**/api/**/*.go'
+            - '!**/server/etcdserver/api/**/*.go'
+          deny:
+            - pkg: go.etcd.io/etcd/api/v3$
```

Comment:
$ at the end is unique. What's it purpose?

Reply:
The problem is the next import: https://github.com/etcd-io/etcd/blob/4601818f511478980725a215e814e56fb8ee31ef/api/etcdserverpb/raft_internal_stringer_test.go#L22

If we remove the `$`, it fails:
```bash
% (cd api && 'golangci-lint' 'run' '--config' '/home/ivan/Code/Personal/etcd/etcd/tools/.golangci.yaml' '--show-stats=false' './...')
FAIL: (code:1):                                                     
  % (cd api && 'golangci-lint' 'run' '--config' '/home/ivan/Code/Personal/etcd/etcd/tools/.golangci.yaml' '--show-stats=false' './...')
../api/etcdserverpb/raft_internal_stringer_test.go:22:2: import 'go.etcd.io/etcd/api/v3/etcdserverpb' is not allowed from list 'api' (depguard)
        pb "go.etcd.io/etcd/api/v3/etcdserverpb"                    
        ^
```

Reply:
> Have you tested the rules? I tried but in my 3 tries I always hit import cycle.

This is actually another way to test it. Remove the dollar sign, and run `make verify-lint`.